addons/misra.py

```diff
@@ -1611,6 +1611,14 @@ def misra_17_8(self, data):
             if var and var.isArgument:
                 self.reportError(token, 17, 8)
 
+    def misra_18_4(self, data):
+        for token in data.tokenlist:
+            if not token.str in {'+', '-', '+=', '-='}:
```

Comment:
what strange dict is this? use a tuple instead:

    if not token.str in ('+', '-', '+=', '-=')

Reply:
This is a set definition:
```
In [1]: type({'+', '-', '+=', '-='})
Out[1]: set
```

Yes, I can use tuple instead.

Reply:
hmm sorry my mistake. yes please use a tuple instead.